compiler/src/dotty/tools/dotc/core/Hashable.scala

```diff
@@ -1,17 +1,19 @@
 package dotty.tools.dotc
 package core
 
+import scala.language.{unsafeNulls => _}
+
 import Types._
 import scala.util.hashing.{ MurmurHash3 => hashing }
 import annotation.tailrec
 
 object Hashable {
 
   /** A null terminated list of BindingTypes. We use `null` here for efficiency */
-  class Binders(val tp: BindingType, val next: Binders)
```

Comment:
Since `Binders` is exclusively used in the form `Binders | Null` it's better to refactor:
```scala
class SomeBinder(val tp: BindingType, val next: Binders)
type Binders = SomeBinder | Null
```

Reply:
Changed both `Binders` and `BindersPair`